examples/vector_databases/neon/neon-postgres-vector-search-pgvector.ipynb

```diff
@@ -0,0 +1,538 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prerequisites\n",
+    "\n",
+    "Before you begin, ensure that you have the following:\n",
+    "\n",
+    "1. A Neon Postgres database. You can create an account and set up a project with a ready-to-use `neondb` database in a few simple steps. For instructions, see [Sign up](https://neon.tech/docs/get-started-with-neon/signing-up) and [Create your first project](https://neon.tech/docs/get-started-with-neon/setting-up-a-project).\n",
+    "2. A connection string for your Neon database. You can copy it from the **Connection Details** widget on the Neon **Dashboard**. See [Connect from any application](https://neon.tech/docs/connect/connect-from-any-app).\n",
+    "3. The `pgvector` extension. Install the extension in Neon by running `CREATE EXTENSION vector;`. For instructions, see [Enable the pgvector extension](https://neon.tech/docs/extensions/pgvector#enable-the-pgvector-extension). \n",
+    "4. Your [OpenAI API key](https://platform.openai.com/account/api-keys).\n",
+    "5. Python and `pip`."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Install required modules\n",
+    "\n",
+    "This notebook requires the `openai`, `psycopg2`, `pandas`, `wget`, and `python-dotenv` packages. You can install them with `pip`:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! pip install openai psycopg2 pandas wget python-dotenv"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Prepare your OpenAI API key\n",
+    "\n",
+    "An OpenAI API key is required to generate vectors for documents and queries.\n",
+    "\n",
+    "If you do not have an OpenAI API key, obtain one from https://platform.openai.com/account/api-keys.\n",
+    "\n",
+    "Add the OpenAI API key as an operating system environment variable. Name the variable `OPENAI_API_KEY`.\n",
+    "\n",
+    "For information about configuring your OpenAI API key as an environment variable, refer to [Best Practices for API Key Safety](https://help.openai.com/en/articles/5112595-best-practices-for-api-key-safety)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Test your OpenAPI key"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "OPENAI_API_KEY is ready\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Test to ensure that your OpenAI API key is defined as an environment variable\n",
+    "# If you run this notebook locally, you may have to reload the terminal and the notebook to make the environment available\n",
+    "\n",
+    "import os\n",
+    "\n",
+    "if os.getenv(\"OPENAI_API_KEY\") is not None:\n",
```

Comment:
alternatively, you could use `getpass` for an easier way to get the secret if it's not defined in the env

Reply:
Added a getpass option that can be used if an environment variable is not set.